devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy

```diff
@@ -411,40 +477,44 @@ private basicSetCmd(val) {
 
 private switchMultilevelSetCmd(level, duration) {
     def levelVal = validateRange(level, 99, 0, 99)
-
     def durationVal = validateRange(duration, 1, 0, 100)
-
     return secureCmd(zwave.switchMultilevelV3.switchMultilevelSet(dimmingDuration: durationVal, value: levelVal))
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
     logTrace "VersionReport: ${cmd}"
-
     def subVersion = String.format("%02d", cmd.applicationSubVersion)
     def fullVersion = "${cmd.applicationVersion}.${subVersion}"
-
     sendEvent(name: "firmwareVersion",  value:fullVersion, displayed: true, type:  null)
     return []
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
-    logTrace "${cmd}"
+    logTrace "BasicReport: ${cmd}"
     sendSwitchEvents(cmd.value, "physical")
     return []
 }
 
 def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd) {
-    logTrace "${cmd}"
+    logTrace "SwitchMultilevelReport: ${cmd}"
     sendSwitchEvents(cmd.value, "digital")
     return []
 }
 
 private sendSwitchEvents(rawVal, type) {
     def switchVal = rawVal ? "on" : "off"
-
     sendEvent(name: "switch",  value:switchVal, displayed: true, type: type)
-
     if (rawVal) {
         sendEvent(name: "level",  value:rawVal, displayed: true, type: type, unit:"%")
     }
-}
\ No newline at end of file
+}
+
+private isButtonAvailable() {
```

Comment:
@sky-nie Since You don't support a child device anymore, is this function still needed (and all related device settings) ?

Reply:
I keep the function "isButtonAvailable" because parts of the preferences's items  have difference default values.